kernel/nano_core/src/uefi.rs

```diff
@@ -0,0 +1,67 @@
+use crate::{early_setup, nano_core, try_exit};
+use boot_info::uefi::STACK_SIZE;
+use bootloader_api::{config::Mapping, BootloaderConfig};
+use core::arch::asm;
+use memory::VirtualAddress;
+
+#[used]
+#[link_section = ".bootloader-config"]
+pub static __BOOTLOADER_CONFIG: [u8; BootloaderConfig::SERIALIZED_LEN] = {
+    let mut config = BootloaderConfig::new_default();
+    config.mappings.physical_memory = Some(Mapping::Dynamic);
+    config.mappings.page_table_recursive =
+        Some(Mapping::FixedAddress(0o177777_776_000_000_000_0000));
+    config.kernel_stack_size = STACK_SIZE as u64;
+    config.serialize()
+};
+
+#[naked]
+#[no_mangle]
+#[link_section = ".init.text"]
+pub unsafe extern "C" fn _start(boot_info: &'static mut bootloader_api::BootInfo) {
+    asm!(
+        // First argument  (rdi): a reference to the boot info (passed through).
+        // Second argument (rsi): the top of the double fault handler stack.
+        "mov rsi, rsp",
+        // Set the kernel stack pointer to the page before the double fault stack.
+        //
+        // +------------+--------------+--------------------+
+        // | guard page | kernel stack | double fault stack |
+        // +------------+--------------+--------------------+
+        // ^                           ^                    ^
+        // |                           |                   rsi (double_fault_stack)
+        // |                          rsp
+        // kernel_stack_start
+        //
+        // where the guard page and double fault stack are both one page, and the kernel stack is
+        // KERNEL_STACK_SIZE_IN_PAGES pages.
+        //
+        // NOTE: Stacks grow downwards e.g. the kernel stack pointer will grow towards the guard
+        // page.
+        "sub rsp, 4096",
+        "call rust_entry",
+        "jmp KEXIT",
+        options(noreturn),
+    );
+}
+
+#[no_mangle]
+unsafe extern "C" fn rust_entry(
```

Comment:
please separate this into an unsafe function (which will deref the boot into pointer) and a safe function (called by the unsafe one) that calls the `nano_core` fn

Reply:
As I mentioned above, we don't need to use raw pointers. The function is still unsafe, though as `double_fault_stack` has to be pointing to the end of a mapped page.

I've added the `unsafe_op_in_unsafe_fn` lint to the `nano_core` crate, which means that the unsafe operations in an unsafe function are clearly marked. Is the lint good enough?